### PR TITLE
[BUG] Fix Benchmarking.run() silently overwriting results for same-class estimators

### LINKED ISSUE #577: [BUG] Benchmarking.run() silently overwrites results when estimators share class names - add labels parameter
**Describe the bug**

`Benchmarking.run()` uses `estimator.__class__.__name__` as the dictionary key for results (line 132). When two estimators of the **same class** but with **different hyperparameters** are benchmarked, the second silently overwrites the first in the results dict, and only the last estimator's scores appear in the final DataFrame.

**To Reproduce**

```python
from sklearn.dummy import DummyClassifier
from sklearn.metrics import accuracy_score
from sklearn.model_selection import PredefinedSplit
import numpy as np
from pyaptamer.benchmarking._base import Benchmarking
X = np.array([[1, 0], [0, 1], [1, 1], [0, 0]] * 5)
y = np.array([1, 0, 1, 0] * 5)
test_fold = np.ones(len(y)) * -1
test_fold[-4:] = 0
cv = PredefinedSplit(test_fold)
clf1 = DummyClassifier(strategy="most_frequent")
clf2 = DummyClassifier(strategy="stratified")
bench = Benchmarking(
    estimators=[clf1, clf2],
    metrics=[accuracy_score],
    X=X, y=y, cv=cv,
)
results = bench.run()
print(results)
# Only shows ONE "DummyClassifier" row - clf1's results are lost!
```

**Expected behavior**

Both estimators should appear in the results with distinct names. The fix should:

1. Add an optional labels parameter to __init__() so users can provide custom names.
2. When labels=None, auto-detect duplicate class names and append _1, _2 suffixes.
3. Be fully backward compatible - existing code without labels still works.

**Additional context**

This is distinct from issue #543 (which only reported detection). This issue proposes a concrete fix with the labels parameter and auto-deduplication logic.

**Versions**
pyaptamer 0.1.0a1
<details>

</details>

### PR DESCRIPTION
**Pull request by Vaishnav88sk**
Add optional labels parameter to Benchmarking.__init__() to let users distinguish same-class estimators. When labels=None (default), auto-generates unique labels by appending _1, _2 suffixes for duplicates.

This prevents silent overwrite of results when benchmarking multiple estimators of the same class with different hyperparameters.

#### Reference Issues/PRs
Fixes #577 

#### What does this implement/fix? Explain your changes.

Fixes a bug where `Benchmarking.run()` silently overwrites results when multiple estimators share the same class name (e.g., two `DummyClassifier` instances with different hyperparameters).
**Root cause:** Line 132 used `estimator.__class__.__name__` as the dict key, so duplicate class names caused overwrite.
**Fix:**
1. Added optional `labels: list[str] | None` parameter to `Benchmarking.__init__()`
2. When `labels=None` (default), auto-generates unique labels via `_generate_labels()` - appends `_1`, `_2` suffixes for duplicate class names
3. When `labels` is provided, validates length matches estimators count
4. `run()` now uses `self.labels` instead of `__class__.__name__`
**Backward compatible** - existing code without `labels` produces identical results when class names are unique.

#### What should a reviewer concentrate their feedback on?

- [x] Auto-deduplication logic in `_generate_labels()`
- [x] Validation of user-provided labels
- [x] Backward compatibility with existing tests

#### Did you add any tests for the change?

Yes - added `TestBenchmarkingLabels` class to `pyaptamer/benchmarking/tests/test_benchmarking.py` with 7 tests:
- Default label = class name
- Auto-dedup with `_1`, `_2` suffixes
- Custom labels stored correctly
- Wrong-length labels raise `ValueError`
- `run()` results use custom labels
- `run()` results use auto-dedup labels
- Backward compatibility with single estimator

#### Any other comments?
All existing tests remain unchanged and should still pass.

#### PR checklist

- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.
- [x] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks. Install hooks with `pre-commit install`.
  To run hooks independent of commit, execute `pre-commit run --all-files`

### CODE REVIEW

**Comment by Vaishnav88sk** [comment]
@fkiraly @NennoMP

**Comment by Vaishnav88sk** [comment]
@SimonBlanke

**Comment by Vaishnav88sk** [comment]
@siddharth7113

**Comment by siddharth7113** [comment]
Hi @Vaishnav88sk,

As per my current understanding the entire Benchmarking module is not working, until the port of Aptatrans is complete,  It is part of roadmap but I haven't got time to revisit it, and I am unable to merge this before i finish that so this fix can be tested.
